app/jobs/concerns/maintenance_tasks/task_job_concern.rb

```diff
@@ -36,13 +36,30 @@ def build_enumerator(_run, cursor:)
       collection_enum = case collection
       when ActiveRecord::Relation
         enumerator_builder.active_record_on_records(collection, cursor: cursor)
+      when ActiveRecord::Batches::BatchEnumerator
+        if collection.start || collection.finish
+          raise ArgumentError, <<~MSG.squish
+            #{@task.class.name}#collection cannot support
+            a batch enumerator with the "start" or "finish" options.
+          MSG
+        end
+        enumerator = enumerator_builder.active_record_on_batch_relations(
+          collection.relation,
+          cursor: cursor,
+          batch_size: collection.batch_size,
+        )
+        @run.update!(tick_total: enumerator.size)
```

Comment:
Like I mentioned above, it should be possible to avoid calling `ActiveRecordBatchEnumerator#size` which is not in their control.